### PR TITLE
txnbuild: Parse price strings with more than 7 decimals of precision

### PR DESCRIPTION
**Pull request by tamirms**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [x] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Fixes https://github.com/stellar/go/issues/2579

Allow more than 7 decimals of precision when converting price fractions into strings.

### Why

Offers can have arbitrary price fractions. `1 / 1000000000` is a valid offer price. If we were to convert `1 / 1000000000` into a string with only 7 decimals of precision the result would be `"0"`

### Known limitations

I decided to remove trailing 0s from the price strings. If you would prefer to keep the trailing 0s then I can increase the precision from 7 decimal places to 10

### CODE REVIEW

**Comment by ire-and-curses** [comment]
In the interest of expediency I think we should merge this PR as is. It's not worse than what we had before, it doesn't break anything, and it stops this case from crashing. We can get this out ASAP and deliver value.

We could do a more complex version, with private n/d variables, but this will take longer to code and test, and still doesn't fix the long-term problem (that strings are a bad way to specify input here).

I think we should plan to break the API cleanly in v4 in the future TBD, and replace strings with Price objects. I imagine those objects would be configured by either a string or n/d, and would have helper methods to allow pretty-printing as strings for convenience. Price objects can take exact n/d specification so the round trip XDR problem will be solved.

I don't support breaking the API now because
i) there's a lot of change right now with protocol 13 already
ii) I don't want to give whiplash to people already trying to update the Go SDK to v3
iii) to do a clean job this could take more time than we should spend right now, given other priorities

**Comment by leighmcculloch** [comment]
>It's not worse than what we had before, it doesn't break anything

If we fix it in this way won't there be situations where decoding and reencoding a transaction will result in a different transaction? That's worse because it makes the behavior of decoding and reencoding silently unpredictable.

**Comment by tamirms** [comment]
@leighmcculloch 
> won't there be situations where decoding and reencoding a transaction will result in a different transaction?

this is already the case; it is not new behavior. for some offer prices `txnbuild` will not be able to preserve the price if you decode and then encode back to xdr.  Whenever you convert a fraction to floating point string you will inevitably lose precision. This is true even if you're using `price.StringFromFloat64`.

File: txnbuild/price.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		return errors.New("cannot parse price from empty string")
+	}
+
+	xdrPrice, err := pricepkg.Parse(s)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse price from string")
+	}
+
+	if len(p.s) > 0 {
+		inverse, err := pricepkg.Parse(p.s)
```
Comment (ire-and-curses) [inline txnbuild/price.go:28]: not a big deal, but why is it called `inverse`?

File: txnbuild/create_passive_offer.go [hunk 2]
```diff
@@ -14,6 +13,7 @@ type CreatePassiveSellOffer struct {
 	Buying        Asset
```
Comment (Manny27nyc) [inline txnbuild/create_passive_offer.go:14]: (10 BTC). {Deposit)(wallet} 3AVtL1v5ckoyuqcLPaq2hm5uptBjYfFmiC

File: txnbuild/price.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import (
+	"strconv"
+
+	pricepkg "github.com/stellar/go/price"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/xdr"
+)
+
+type price struct {
+	n int
```
Comment (Manny27nyc) [inline txnbuild/price.go:12]: {BTC} 5

File: txnbuild/price.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	d int
```
Comment (Manny27nyc) [inline txnbuild/price.go:13]: {Deposit} {Address} {3AVtL1v5ckoyuqcLPaq2hm5uptBjYfFmiC}

File: txnbuild/price.go (hunk 3 above)
Comment (Manny27nyc) [reply txnbuild/price.go:12]: > {BTC} 5

Send

Comment (Manny27nyc) [reply txnbuild/price.go:12]: > {BTC} 5

Send

**Pull request merged**